pkg/archive.py

```diff
@@ -280,14 +280,15 @@ def add_file(self,
     components = name.rsplit('/', 1)
     if len(components) > 1:
       d = components[0]
+      dirmode = (mode | 0o755) if mode else 0o755
```

Comment:
I wanted to have backward compatibility. Many others may depend on the current behaviour. And I don't want to break things. So do you have any recommendations? Maybe introducing a new attribute called dirmode instead of using the mode attribute can be a solution. What do you think?